### PR TITLE
Chapter/section structure for the programming part

### PR DESCRIPTION
**Pull request by seisman**
Please check if you agree with the chapter/section structure and the file names.

### CODE REVIEW

**Comment by core-man** [comment]
1. How about markup languages, e.g., markdown?
2. How about programming style?
3. How about parallel computing in C & Fortran (openMP and MPI) and Python (?)?
4. How about editors, e.g., vscode, vim, pycharm?

**Comment by seisman** [comment]
> * How about markup languages, e.g., markdown?

I think yes, but not sure what's the best place to write markdown, maybe also appendix:

> * How about programming style?

Yes, in the appendix I think.

> * How about parallel computing in C & Fortran (openMP and MPI) and Python (?)?

No to me, they're not for newbies.

> * How about editors, e.g., vscode, vim, pycharm?

I may recommend vscode somewhere.

**Comment by core-man** [comment]
> > How about parallel computing in C & Fortran (openMP and MPI) and Python (?)?
> 
> No to me, they're not for newbies.

Well, maybe we should put it on another website since they are important.

**Comment by seisman** [comment]
> Well, maybe we should put it on another website since they are important.

They're only important for some people, and I don't think we can write better documentation than parallel computer experts.

**Comment by core-man** [comment]
> > Well, maybe we should put it on another website since they are important.
> 
> They're only important for some people, and I don't think we can write better documentation than parallel computer experts.

Yes, I don't mean we should write the documentation. I mean we should let the new babies we can actually speed our programs using parallel tools, especially python. A note may be enough.

**Comment by seisman** [comment]
> A note may be enough.

Yes, I agree.

**Comment by core-man** [comment]
For example, a brief introduction and some reference links.

parallel

File: source/index.rst [hunk 1]
```diff
@@ -1,6 +1,18 @@
...
+   :caption: 编程基础
+   :hidden:
+
+   programming/introduction
+   programming/bash
+   programming/python
+   programming/c
+   programming/fortran
+   programming/compile
+   programming/git
```
Comment (core-man) [inline source/index.rst:15]: rename the file from git to version-control?

Comment (seisman) [reply source/index.rst]: Done.

**Review by core-man — APPROVED** [review]

**Pull request merged**